### ISSUE TITLE
Add 301s

### ISSUE THREAD

**Issue by mjpost** [acl-org/acl-anthology#178]
We should add redirects for old aclanthology.org pages. This should cover:

- authors (e.g., https://aclanthology.info/people/brian-roark/ → http://www.aclweb.org/anthology/people/b/brian-roark/)
- papers
- venues

**Comment by mbollmann** [comment]
I don't think author pages can consistently be redirected, as the two websites use different systems to disambiguate names that look the same in their "slugified" version; e.g. 

+ https://aclanthology.info/people/geraldine-walther-4b92ae6a-a443-4247-a897-b5b7d7effa37 → http://www.aclweb.org/anthology/people/g/geraldine-walther/
+ https://aclanthology.info/people/hector-martinez-alonso → http://www.aclweb.org/anthology/people/h/hector-martinez-alonso/
+ https://aclanthology.info/people/hector-martinez-alonso-82959cec-fbdb-47a5-adb7-6a855b051edd → http://www.aclweb.org/anthology/people/h/hector-martinez-alonso2/

Adressing #86 will complicate this even further.

For the paper pages, I proposed (via e-mail) to do this within the Ruby app, mainly because I thought deriving the hierarchical URL is a bit more involved. However, if Apache rewrite rules are already linking flat IDs to hierarchical URLs for retrieving the PDFs, surely the same could be done for 301s from aclanthology.info to aclweb.org?

**Comment by mjpost** [comment]
Yes, I agree about author pages. And they are probably less important to redirect, though what brought this to mind yesterday was coming across a high-ranked ACL author page on aclanthology.info when Googling someone.

@martin, what do you think about these approaches to 301s for aclanthology.info?

**Comment by villalbamartin** [comment]
Right now we are not doing any rewrite rules in the server, but we could easily do it for papers and venues, since the redirection format is straightforward.

Do we have a way to consistently obtain the new slug for a person given either the old one or their entry in the database? If there is code around that can do that, we can probably tell our webserver (which at the moment is Nginx, not Apache) to pick the redirection URL from some local web service.

**Comment by mbollmann** [comment]
> Do we have a way to consistently obtain the new slug for a person given either the old one or their entry in the database? 

No, and I don't think that's possible, since the new slugs are generated independently from the current ones or the database.

**Comment by mjpost** [comment]
Just an update here. @villalbamartin will be working on this in the near future. Once this is done, I think we can mark the static rewrite as completed.

**Comment by villalbamartin** [comment]
Apologies that this took so long. I am back for good now, and hopefully a lot more active.

Redirect for papers, events, venues, and main page are now in place. Everything else is still being served from the old system.

I will keep an eye on which pages are not being redirected in our logs, and see if there's something important that we missed. Note also that author pages are _not_ being redirected, because AFAIK we don't really know how to do that.

**Comment by mjpost** [comment]
This is great, thanks, Martín!

**Comment by villalbamartin** [comment]
Even though the issue is closed, here's a small report on what remains to be redirected.

A 24hs analysis of the logs, and after removing most bot traffic, reveals that these are the remaining popular URLs that are still being served by the legacy anthology:

Requests | Address
--- | ---
1145 | /people
603 | /catalog
428 | /pdf
222 | /volumes

Of these, `/people` remains an open problem: once we factor bot traffic back in, we served 21K requests in 24hs. So I'm not sure about pulling the plug on that one yet. I'll look into alternatives.

`/pdf` can already be redirected, and `/volumes` seems to be working well in the new Anthology. If no one has any objection, I will redirect both to the new Anthology.

Finally, out of the 600 requests to `/catalog`, around 500 are from people trying to hack our server, so nothing of value is lost there.

**Comment by mjpost** [comment]
Ah, maybe I was a bit trigger-happy in closing it. Thanks for looking at this so fastidiously.

Redirecting `/pdf` and `/volumes` sounds like a great idea.

For people, what about writing a script that checks if the corresponding URL is present on the new Anthology, and redirecting it if it exists? There may be some wrong mappings, but maybe not, and in any case the precision should be pretty high. I'm also fine to just do a blanket redirect on those pages.

It will be nice to see how these requests wane as the weeks go on. Thanks, Martín.

**Comment by mjpost** [comment]
@villalbamartin, can you check the logs and see if any are still missing? If not we can close this out.

**Comment by villalbamartin** [comment]
Here's the logs from the last two weeks:
  * Most PDFs are being redirected properly. There was one missing URL format responsible for a couple thousand hits. This has now been corrected.
  * We have 2.5K requests for URLs starting with `/volumes/` and requesting a specific type of file (example: https://aclanthology.info/volumes/proceedings-of-the-workshop-on-integer-linear-programming-for-natural-language-processing.acm). These are currently not being redirected because it is not clear what the redirection to a .acm or .ris file should be. The simplest case would be redirecting to the main page of the volume, stripping the extension of the request and letting the visitor figure out which format they want to get.
  * As expected, we have 20K hits for `/people/` in April 2019.

I imagine once we decide what to do with `volumes`, we can close the issue as `done/wontfix`, unless we find someone willing to write a script that redirects from old authors to new ones with decent accuracy.

**Comment by mjpost** [comment]
Can you post a list for `people`? I bet we could automate most of the redirects.

**Comment by villalbamartin** [comment]
Sure, here's the list without duplicates: [people.txt](https://github.com/acl-org/acl-anthology/files/3095543/people.txt)

**Comment by mjpost** [comment]
It looks like `/people` are being redirected. Let's let this redirect until May 1, and then take it down. Sound okay?
